File: Bridge/BridgePattern.py

from abc import ABC, abstractmethod

class Abstraction(ABC):
    def __init__(self, imp):
        self._imp = imp

    @abstractmethod
    def operation(self):
        pass

class ConcreteAbstraction(Abstraction):
    def operation(self):
        self._imp.operation()


class Implementor(ABC):
    @abstractmethod
    def operation(self):
        pass


class ConcreteImplementor1(Implementor):
    def operation(self):
        print("ConcreteImplementor1 operation")


class ConcreteImplementor2(Implementor):
    def operation(self):
        print("ConcreteImplementor2 operation")

c1 = ConcreteImplementor1()
abstraction = ConcreteAbstraction(c1)
abstraction.operation()

File: Bridge/BridgePattern.java
// Abstraction interface
interface Abstraction {
    void operation();
}

// Implementor interface
interface Implementor {
    void operation();
}

// ConcreteImplementor1 class
class ConcreteImplementor1 implements Implementor {
    @Override
    public void operation() {
        System.out.println("ConcreteImplementor1 operation");
    }
}

// ConcreteImplementor2 class
class ConcreteImplementor2 implements Implementor {
    @Override
    public void operation() {
        System.out.println("ConcreteImplementor2 operation");
    }
}

// ConcreteAbstraction class
class ConcreteAbstraction implements Abstraction {
    private Implementor imp;

    public ConcreteAbstraction(Implementor imp) {
        this.imp = imp;
    }

    @Override
    public void operation() {
        imp.operation();
    }
}

// Client code
public class BridgePattern {
    public static void main(String[] args) {
        Implementor c1 = new ConcreteImplementor1();
        Abstraction abstraction = new ConcreteAbstraction(c1);
        abstraction.operation();
    }
}
